stuffer/s2n_stuffer_file.c

```diff
@@ -28,7 +28,9 @@
 
 int s2n_stuffer_recv_from_fd(struct s2n_stuffer *stuffer, const int rfd, const uint32_t len, uint32_t *bytes_written)
 {
+    POSIX_ENSURE_REF(stuffer->blob.data);
```

Comment:
If this precondition was introduced just to avoid erring out on the `read` call, can we instead change line 42 to:

```c
r = (stuffer->blob.data) ? read(rfd, stuffer->blob.data + stuffer->write_cursor, len) : -1;
```

This would not rule out `stuffer->blob.data == NULL` code path entirely, but just make sure that it's correctly handled for `read`.

Reply:
Implemented as suggested.